trakt/shows.go

```diff
@@ -763,14 +763,94 @@ func WatchedShowsProgress() (shows []*ProgressShow, err error) {
 	wg.Wait()
 
 	for _, s := range showsList {
-		if s != nil {
+		if s != nil && !ShowIsHidden("progress_watched", s.Show.IDs.Trakt) {
```

Comment:
That needs to be changed. If you have 100 shows, it will make 100 calls, each will read hidden shows, iterate them. 
You can leave `ShowIsHidden` and add implementation which accepts an array and returns an array or map (map would be less problematic).

Reply:
@elgatito 
good catch, thank you.
i was not sure what exactly you meant so i added 2 implementations of how i see it. please review.